cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedules.java

```diff
@@ -64,4 +64,88 @@ private Schedules() {
   public enum Source {
     STREAM
   }
+
+  /**
+   * Get a builder used to create a schedule.
+   *
+   * @param name the name of the schedule to build
+   * @return a builder used to create a schedule
+   */
+  public static Builder builder(String name) {
+    return new Builder(name);
+  }
+
+  /**
+   * Builder used to create a schedule.
+   */
+  public static class Builder {
+    private final String name;
+    private String description;
+    private Integer maxConcurrentRuns;
+
+    private Builder(String name) {
+      this.name = name;
+      this.description = "";
+    }
+
+    /**
+     * Set the schedule description.
+     *
+     * @param description the schedule description
+     * @return the Builder
+     */
+    public Builder setDescription(String description) {
+      this.description = description;
+      return this;
+    }
+
+    /**
+     * Set the maximum number of concurrent runs for the schedule. Prior to running the program,
+     * the scheduler will check how many active runs of the scheduled program exist.
+     * An active run is one that is not completed, failed, or killed. This includes suspended runs.
+     * If that number is greater than or equal to the max, the scheduler will not run the program.
+     * This number does not include runs of the program that were started outside of this schedule.
+     * For example, if the threshold is set to 1, the scheduler will
+     * only run the program if there are no current runs.
+     *
+     * If no threshold is set, there will be no limit on the number of concurrent runs.
+     *
+     * @param max skip the scheduled run if the number of concurrent program runs is above this threshold
+     * @return the Builder
+     */
+    public Builder setMaxConcurrentRuns(int max) {
+      if (max < 1) {
+        throw new IllegalArgumentException("max concurrent runs must be at least 1.");
+      }
+      this.maxConcurrentRuns = max;
+      return this;
+    }
+
+    /**
+     * Create a time schedule based on the specified cron expression.
+     *
+     * @param cronExpression the cron expression specifying when the scheduler should attempt to start a run
+     * @return the time schedule
+     */
+    public Schedule createTimeSchedule(String cronExpression) {
```

Comment:
I think they should, but wanted to do it separately since they're used all over the place in tests.